server.go

```diff
@@ -183,7 +184,9 @@ func (u *Upgrader) Upgrade(w http.ResponseWriter, r *http.Request, responseHeade
 	}
 
 	if brw.Reader.Buffered() > 0 {
-		netConn.Close()
+		if err := netConn.Close(); err != nil {
+			log.Printf("websocket: failed to close network connection: %v", err)
```

Comment:
Failure to close network connection is secondary to early write by client.  Write to application's stderr is unfriendly.  Revert to original or use errors.Join.